File: assignment_1/app/src/main/java/edu/fullerton/cpsc411/assignment_1/MainActivity.java
package edu.fullerton.cpsc411.assignment_1;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import static java.lang.Math.PI;
import static java.util.Locale.US;


public class MainActivity extends AppCompatActivity implements TextWatcher {
    TextView result;
    EditText input1, input2 ;
    Double  final_output;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        input1 = findViewById(R.id.mib_input);
        input2 = findViewById(R.id.mbps_input);
        result = findViewById(R.id.output);

        input1.addTextChangedListener(this);
        input2.addTextChangedListener(this);


    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {


        if (!(input1.getText().toString()).equals("")  && !(input2.getText().toString()).equals("")  ) {

            final_output =cal(Double.valueOf(input1.getText().toString()), Double.valueOf(input2.getText().toString()));

            String f = roundOff(final_output).toString()+ " Seconds";

            result.setText(f);
        }

        else {
            result.setText("0");
        }



    }

    @Override
    public void afterTextChanged(Editable s) {

    }

    Double cal (Double mib , Double mbp){

        return (mib * Math.pow(2, 20) * 8) / (mbp * Math.pow(10, 6)) ;
    }

    String roundOff(Double final_output) {

        return String.format(US,"%.1f", final_output);

    }




}

File: assignment_2/app/src/main/java/edu/fullerton/cpsc411/assignment_1/MainActivity.kt
package edu.fullerton.cpsc411.assignment_1
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.EditText
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.Locale.US

internal fun cal(mib: Double?, mbp: Double?): Double? {

    return mib!! * Math.pow(2.0, 20.0) * 8.0 / (mbp!! * Math.pow(10.0, 6.0))
}

internal fun roundOff(final_output: Double?): String {

    return String.format(US, "%.1f", final_output)

}

class MainActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fun TextView.finalOutput(){
            val input1 = mib_input.text
            val input2 = mbps_input.text
            if(input1.isNotEmpty() && input2.isNotEmpty()){
                val finalCal = cal(java.lang.Double.valueOf(mib_input.text.toString()), java.lang.Double.valueOf(mbps_input.text.toString()))

                val f = roundOff(finalCal) + " Seconds"

                this.text = f
            }
            else{
                this.text = "0"

            }
        }

        fun EditText.setWatcher() {
            this.addTextChangedListener(object : TextWatcher {
                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
                }

                override fun afterTextChanged(s: Editable?) {
                }

                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                    output.finalOutput()
                }

            })
        }

        mbps_input.setWatcher()
        mib_input.setWatcher()

    }
}










